docs/rules/func-names.md

```diff
@@ -45,6 +46,34 @@ Foo.prototype.bar = function bar() {};
 }())
 ```
 
+### as-needed
+
+ECMAScript 6 introduced a name property on all functions. The value of name is determined by evaluating the code around the function to see if a name can be inferred. For example, a function assigned to a variable will automatically have a name property equal to the name of the variable. The value of name is then used in stack traces for easier debugging.
```

Comment:
Sorry, one last nitpick, can you put "name" in backticks so it's obvious that it's a property (each sentence has at least one reference that should be in backticks).